tests/integration_tests/test_visualization.py

```diff
@@ -326,6 +330,94 @@ def test_visualization_compare_classifiers_from_prob_csv_output_saved(
             print("Error: %s - %s." % (e.filename, e.strerror))
 
 
+def test_visualization_compare_classifiers_from_prob_csv_output_saved_api(
+        csv_filename
+):
+    """Ensure pdf and png figures from the experiments can be saved.
+
+    Probabilities are loaded from csv file.
+    :param csv_filename: csv fixture from tests.fixtures.filenames.csv_filename
+    :return: None
+    """
+    input_features = [
+        text_feature(vocab_size=10, min_len=1, representation='sparse'),
+        category_feature(vocab_size=10)
+    ]
+    output_features = [category_feature(vocab_size=2, reduce_input='sum')]
+
+    # Generate test data
+    rel_path = generate_data(input_features, output_features, csv_filename)
+    input_features[0]['encoder'] = 'parallel_cnn'
+
+    config = {
+        'input_features': input_features,
+        'output_features': output_features,
+        'combiner': {
+            'type': 'concat',
+            'fc_size': 14
+        },
+        'training': {'epochs': 2}
+    }
+
+    model = LudwigModel(
+        config,
+        logging_level=logging.WARN
+    )
+
+    (
+        _,
+        preprocessed_data,
+        exp_dir_name
+    ) = model.train(dataset=rel_path)
+
+    _, _, test_set, metadata = preprocessed_data
+
+    model.predict(
+        dataset=test_set,
+        skip_save_predictions=False,
+        output_directory=exp_dir_name
+    )
+
+    output_feature_name = config['output_features'][0][NAME]
+    probability = os.path.join(exp_dir_name, '{}_probabilities.csv').format(
```

Comment:
obtain the probabilities from the tensors obtain from predict instead of reading them from disk

Reply:
done